emcc.py

```diff
@@ -1197,6 +1197,13 @@ def phase_setup(options, state, newargs, settings_map):
     diagnostics.warning('deprecated', 'RUNTIME_LINKED_LIBS is deprecated; you can simply list the libraries directly on the commandline now')
     newargs += settings.RUNTIME_LINKED_LIBS
 
+  def default_setting(name, new_default):
+    if name not in settings_map:
+      setattr(settings, name, new_default)
+
+  if settings.STRICT:
+    default_setting('DEFAULT_TO_CXX', 0)
```

Comment:
Why move this here? I can't see uses of it between this place and the old location.

Reply:
DEFAULT_TO_CXX is used during the compile phase here: https://github.com/emscripten-core/emscripten/blob/2126d3c1a64c9db2ab390b2c09fd91ef41d05b24/emcc.py#L2320-L2329... so setting in latest during `phase_linker_setup` is ineffective.

Reply:
I should make that a separate change though.

Reply:
https://github.com/emscripten-core/emscripten/pull/14847